deepmd/infer/deep_property.py

```diff
@@ -0,0 +1,138 @@
+    """
+
+    @property
+    def output_def(self) -> ModelOutputDef:
+        """Get the output definition of this model."""
+        return ModelOutputDef(
+            FittingOutputDef(
+                [
+                    OutputVariableDef(
+                        "property",
+                        shape=[-1],
+                        reduciable=True,
+                        atomic=True,
+                    ),
+                ]
+            )
+        )
+
+    @property
+    def task_dim(self) -> int:
+        """Get the output dimension."""
+        return self.get_task_dim()
+
+    def eval(
+        self,
+        coords: np.ndarray,
+        cells: Optional[np.ndarray],
+        atom_types: Union[List[int], np.ndarray],
+        atomic: bool = False,
+        fparam: Optional[np.ndarray] = None,
+        aparam: Optional[np.ndarray] = None,
+        mixed_type: bool = False,
+        **kwargs: Dict[str, Any],
+    ) -> Tuple[np.ndarray, ...]:
+        """Evaluate properties. If atomic is True, also return atomic property.
+
+        Parameters
+        ----------
+        coords : np.ndarray
+            The coordinates of the atoms, in shape (nframes, natoms, 3).
+        cells : np.ndarray
+            The cell vectors of the system, in shape (nframes, 9). If the system
+            is not periodic, set it to None.
+        atom_types : List[int] or np.ndarray
+            The types of the atoms. If mixed_type is False, the shape is (natoms,);
+            otherwise, the shape is (nframes, natoms).
+        atomic : bool, optional
+            Whether to return atomic property, by default False.
+        fparam : np.ndarray, optional
+            The frame parameters, by default None.
+        aparam : np.ndarray, optional
+            The atomic parameters, by default None.
+        mixed_type : bool, optional
+            Whether the atom_types is mixed type, by default False.
+        **kwargs : Dict[str, Any]
+            Keyword arguments.
+
+        Returns
+        -------
+        property
+            The properties of the system, in shape (nframes, num_tasks).
+        """
+        (
+            coords,
+            cells,
+            atom_types,
+            fparam,
+            aparam,
+            nframes,
+            natoms,
+        ) = self._standard_input(coords, cells, atom_types, fparam, aparam, mixed_type)
+        results = self.deep_eval.eval(
+            coords,
+            cells,
+            atom_types,
+            atomic,
+            fparam=fparam,
+            aparam=aparam,
+            **kwargs,
+        )
+        atomic_property = results["property"].reshape(
+            nframes, natoms, self.get_task_dim()
+        )
+        property = results["property_redu"].reshape(nframes, self.get_task_dim())
+
+        if atomic:
+            return (
+                property,
+                atomic_property,
+            )
+        else:
+            return (property,)
```

Comment:
is the property normalized by 1/N if the system is intensive? if so where have you implemented it?

Reply:
![20240628-110219](https://github.com/deepmodeling/deepmd-kit/assets/100290172/64ec7b1b-29f6-4c5b-bace-f0a0090d0982)
The property is  normalized by 1/N if the system is intensive. It has been done in function `self.deep_eval.eval`, specifically in `PropertyModel` module. So we do not need to normalize it in `DeepProperty`.

Reply:
It should be handled by the model, just like all the reducible outputs are handled by the model.